adapters/handlers/grpc/v1/mapping.go

```diff
@@ -192,23 +182,222 @@ func NewObject[P schema.PropertyInterface](v map[string]interface{}, parent P, s
+}
+
+func (m *Mapper) newListValueDate(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_DateValues{DateValues: &pb.DateValues{Values: []string{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[string](v, schema.DataTypeDateArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_DateValues{DateValues: &pb.DateValues{Values: values}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewDateValue(v)
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
 }
 
-// NewList constructs a ListValue from a general-purpose Go slice.
-// The slice elements are converted using NewValue.
-func NewObjectList[P schema.PropertyInterface](v []interface{}, parent P, selectProp search.SelectProperty) (*pb.ListValue, error) {
+func (m *Mapper) newListValueNumber(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_NumberValues{NumberValues: &pb.NumberValues{Values: []float64{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[float64](v, schema.DataTypeNumberArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_NumberValues{NumberValues: &pb.NumberValues{Values: values}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewNumberValue(v)
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueInt(v interface{}) (*pb.Value, error) {
+	var listValue *pb.ListValue
+	if _, ok := v.([]interface{}); ok {
+		if m.uses125 {
+			listValue = &pb.ListValue{Kind: &pb.ListValue_IntValues{IntValues: &pb.IntValues{Values: []int64{}}}}
+		} else {
+			listValue = &pb.ListValue{Values: []*pb.Value{}}
+		}
+	} else {
+		values, err := parseArray[float64](v, schema.DataTypeIntArray)
+		if err != nil {
+			return nil, err
+		}
+		if m.uses125 {
+			parsedValues := make([]int64, len(values))
+			for i, v := range values {
+				parsedValues[i] = int64(v)
+			}
+			listValue = &pb.ListValue{Kind: &pb.ListValue_IntValues{IntValues: &pb.IntValues{Values: parsedValues}}}
+		} else {
+			x := make([]*pb.Value, len(values))
+			for i, v := range values {
+				x[i] = NewIntValue(int64(v))
+			}
+			listValue = &pb.ListValue{Values: x}
+		}
+	}
+	return &pb.Value{Kind: &pb.Value_ListValue{ListValue: listValue}}, nil
+}
+
+func (m *Mapper) newListValueString(v interface{}) (*pb.Value, error) {
```

Comment:
String and text datatypes are identical and string is deprecated - AFAIK there was a difference in the tokenization settings which can now be controlled through a separate field

Reply:
I felt that we should still handle returning them. Do you think its safe to remove from the gRPC API now?

Reply:
I think doing
```
	case schema.DataTypeStringArray, schema.DataTypeTextArray:
		return m.newListValueText(v)
```

should work